File: src/test/java/com/codingbat/warmup1/ParrotTroubleTest.java
package com.codingbat.warmup1;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class ParrotTroubleTest {
  @Test
  public void inTroubleIfParrotTalkingAtSixHours() {
    assertTrue(ParrotTrouble.parrotTrouble(true, 6));
  }

  @Test
  public void notInTroubleIfParrotTalkingAtSevenHours() {
    assertFalse(ParrotTrouble.parrotTrouble(true, 7));
  }

  @Test
  public void notInTroubleIfParrotNotTalkingAtSixHours() {
    assertFalse(ParrotTrouble.parrotTrouble(false, 6));
  }

  @Test
  public void inTroubleIfParrotTalkingAtTwentyOneHours() {
    assertTrue(ParrotTrouble.parrotTrouble(true, 21));
  }

  @Test
  public void notInTroubleIfParrotNotTalkingAtTwentyOneHours() {
    assertFalse(ParrotTrouble.parrotTrouble(false, 21));
  }

  @Test
  public void notInTroubleIfParrotNotTalkingAtTwentyHourse() {
    assertFalse(ParrotTrouble.parrotTrouble(false, 20));
  }

  @Test
  public void inTroubleIfParrotTalkingAtTwentyThreeHours() {
    assertTrue(ParrotTrouble.parrotTrouble(true, 23));
  }

  @Test
  public void notInTroubleIfParrotTalkingAtTwentyThreeHours() {
    assertFalse(ParrotTrouble.parrotTrouble(false, 23));
  }

  @Test
  public void notInTroubleIfParrotTalkingAtTwentyHours() {
    assertFalse(ParrotTrouble.parrotTrouble(true, 20));
  }

  @Test
  public void notInTroubleIfParrotNotTalkingAtTwelveHours() {
    assertFalse(ParrotTrouble.parrotTrouble(false, 12));
  }
}

File: src/test/kotlin/com/codingbat/warmup1/ParrotTroubleTest.kt
package com.codingbat.warmup1

import org.junit.Test
import kotlin.test.assertTrue
import kotlin.test.assertFalse

class ParrotTroubleTestKt {
  @Test
  fun inTroubleIfParrotTalkingAtSixHours() {
    assertTrue(parrotTrouble(true, 6))
  }

  @Test
  fun notInTroubleIfParrotTalkingAtSevenHours() {
    assertFalse(parrotTrouble(true, 7))
  }

  @Test
  fun notInTroubleIfParrotNotTalkingAtSixHours() {
    assertFalse(parrotTrouble(false, 6))
  }

  @Test
  fun inTroubleIfParrotTalkingAtTwentyOneHours() {
    assertTrue(parrotTrouble(true, 21))
  }

  @Test
  fun notInTroubleIfParrotNotTalkingAtTwentyOneHours() {
    assertFalse(parrotTrouble(false, 21))
  }

  @Test
  fun notInTroubleIfParrotNotTalkingAtTwentyHourse() {
    assertFalse(parrotTrouble(false, 20))
  }

  @Test
  fun inTroubleIfParrotTalkingAtTwentyThreeHours() {
    assertTrue(parrotTrouble(true, 23))
  }

  @Test
  fun notInTroubleIfParrotTalkingAtTwentyThreeHours() {
    assertFalse(parrotTrouble(false, 23))
  }

  @Test
  fun notInTroubleIfParrotTalkingAtTwentyHours() {
    assertFalse(parrotTrouble(true, 20))
  }

  @Test
  fun notInTroubleIfParrotNotTalkingAtTwelveHours() {
    assertFalse(parrotTrouble(false, 12))
  }
}
